src/FindTranslocations.cpp

```diff
@@ -306,105 +265,47 @@ int main(int argc, char *argv[]) {
-			if ( !(vm.count("coverage") and vm.count("insert") and vm.count("insert-std")) ){
-				DEFAULT_CHANNEL << "--coverage , --insert, and --insert-std must be all specified at the same time. Please specify all three or let the program compute the numbers (Bam file wil be read twice).\n";
-				return 2;
-			}
-		}
-
-		if (vm.count("coverage")) {
+		LibraryStatistics library;
+		size_t start = time(NULL);
+		library = computeLibraryStats(alignmentFile, genomeLength, max_insert,40, outtie,minimum_mapping_quality,outputFileHeader);
+		printf ("library stats time consumption= %lds\n", time(NULL) - start);
+		coverage   = library.C_A;
+		if(vm.count("coverage")){
 			coverage    = vm["coverage"].as<float>();
-			meanInsert  = vm["insert"].as<float>();
-			insertStd   = vm["insert-std"].as<float>();
-
-		 }else {
-			LibraryStatistics library;
-			size_t start = time(NULL);
-			library = computeLibraryStats(alignmentFile, genomeLength, max_insert,min_insert, outtie,minimum_mapping_quality);
-			printf ("library stats time consumption= %lds\n", time(NULL) - start);
-			coverage   = library.C_A;
-			meanInsert = library.insertMean;
-			insertStd  = library.insertStd;
-			//update the max_insert
-			if(vm.count("auto")){
-				max_insert =meanInsert+4*insertStd;
-				outtie=library.mp;
-				if(outtie == true){
-					cout << "auto-config orientation: outtie" << endl;
-				}else{
-					cout << "auto-config orientation: inne" << endl;
-				}
-			}
+		}
+		outtie=library.mp;
+		if(outtie == true){
+			cout << "auto-config orientation: outtie" << endl;
+		}else{
+			cout << "auto-config orientation: inne" << endl;
+		}
+		meanInsert = library.insertMean;
+		insertStd  = library.insertStd;
+		if(outtie == false){
+			max_insert =meanInsert+1.5*insertStd;
+		}else{
+			max_insert =meanInsert+4*insertStd;
+		}
 
+        min_insert = meanInsert/2; 
+		if(vm.count("max-insert")){
+			max_insert  = vm["insert"].as<float>();
 		}
-		
+
         int ploidy = 2;
         if(vm.count("ploidy")){
             ploidy = vm["ploidy"].as<int>();
         }
-		size_t start = time(NULL);
-		Cov *calculateCoverage;
-		calculateCoverage = new Cov();
-		calculateCoverage -> coverageMain(alignmentFile,indexFile,"",outputFileHeader,coverage,contig2position,4,500);
-		printf ("time consumption of the coverage computation= %lds\n", time(NULL) - start);
-
 		StructuralVariations *FindTranslocations;
 		FindTranslocations = new StructuralVariations();
-		FindTranslocations -> findTranslocationsOnTheFly(alignmentFile, min_insert, max_insert, outtie, minimum_mapping_quality, minimumSupportingPairs, coverage, meanInsert, insertStd, outputFileHeader, indexFile,contigsNumber,ploidy);
-
+		FindTranslocations -> findTranslocationsOnTheFly(alignmentFile, min_insert, max_insert, outtie, minimum_mapping_quality, minimumSupportingPairs, coverage, meanInsert, insertStd, outputFileHeader, indexFile,contigsNumber,ploidy,library.readLength);
```

Comment:
suggestion: maybe it is better if you create an options object, and you pass this object to the functions. In this way it becomes easier to change the options, add/remove them. Just keep this in mind for the next improved version